docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,118 @@
+# Capstone Architecture overview
+
+## Architecture of Capstone
+
```

Comment:
@kabeor @aquynh could you please help us with this part? So that @Rot127 could focus on addressing the feedback and other architectures apart from ARM

Reply:
Tracked it above. We can open an issue after this is merged.